node/full_node_test.go

```diff
@@ -73,30 +69,58 @@ func TestMempoolDirectly(t *testing.T) {
 		assert.NoError(node.Stop())
 	}()
 
-	pid, err := peer.IDFromPrivateKey(anotherKey)
-	require.NoError(err)
-	err = node.Mempool.CheckTx([]byte("tx1"), func(r *abci.Response) {}, mempool.TxInfo{
-		SenderID: node.mempoolIDs.GetForPeer(pid),
-	})
-	require.NoError(err)
-	err = node.Mempool.CheckTx([]byte("tx2"), func(r *abci.Response) {}, mempool.TxInfo{
-		SenderID: node.mempoolIDs.GetForPeer(pid),
-	})
-	require.NoError(err)
-	time.Sleep(100 * time.Millisecond)
-	err = node.Mempool.CheckTx([]byte("tx3"), func(r *abci.Response) {}, mempool.TxInfo{
-		SenderID: node.mempoolIDs.GetForPeer(pid),
-	})
-	require.NoError(err)
-	err = node.Mempool.CheckTx([]byte("tx4"), func(r *abci.Response) {}, mempool.TxInfo{
-		SenderID: node.mempoolIDs.GetForPeer(pid),
-	})
+	pid := getPeerID(anotherKey, require)
+
+	checkTransactions(node, pid, require)
+
+	checkMempoolSize(node, require)
+}
+
+func setupMockApplication() *mocks.Application {
+	app := &mocks.Application{}
+	app.On(InitChain, mock.Anything).Return(abci.ResponseInitChain{})
+	app.On(CheckTx, mock.Anything).Return(abci.ResponseCheckTx{})
+	return app
+}
+
+func generateKeys() (crypto.PrivKey, crypto.PrivKey) {
+	key, _, _ := crypto.GenerateEd25519Key(rand.Reader)
+	signingKey, _, _ := crypto.GenerateEd25519Key(rand.Reader)
+	return key, signingKey
+}
```

Comment:
you don't need this. This is essentially
```suggestion
func generateKeys() (crypto.PrivKey, crypto.PrivKey) {
	return generateSingleKey(), generateSingleKey()
}
```
which is an unnecessary helper function